src/engraving/libmscore/part.cpp

```diff
@@ -241,12 +268,13 @@ void Part::setStaves(int n)
         qDebug("Part::setStaves(): remove staves not implemented!");
         return;
     }
+
     int staffIdx = score()->staffIdx(this) + ns;
     for (int i = ns; i < n; ++i) {
-        Staff* staff = new Staff(score());
-        staff->setPart(this);
+        Staff* staff = createStaff(score(), this);
         _staves.push_back(staff);
-        score()->staves().insert(staffIdx, staff);
+        const_cast<QList<Staff*>&>(score()->staves()).insert(staffIdx, staff);
```

Comment:
Could we avoid a const_cast here and export a "insertStaff" function from Score? (much as I'd hate to make that interface any bigger!)

Reply:
This is a hack. I have absolutely no idea what is going on in this method  :) It was really easy to apply my changes to all libmscore, but not in this method. For some reason, insertStaff breaks it, so I decided to left it as it is